src/core/services/theming/theming.js

```diff
@@ -154,9 +154,32 @@ var LIGHT_FOREGROUND = {
 var DARK_SHADOW = '1px 1px 0px rgba(0,0,0,0.4), -1px -1px 0px rgba(0,0,0,0.4)';
 var LIGHT_SHADOW = '';
 
-var DARK_CONTRAST_COLOR = colorToRgbaArray('rgba(0,0,0,0.87)');
-var LIGHT_CONTRAST_COLOR = colorToRgbaArray('rgba(255,255,255,0.87)');
-var STRONG_LIGHT_CONTRAST_COLOR = colorToRgbaArray('rgb(255,255,255)');
+var DARK_CONTRAST_COLORS = {
+  primary: colorToRgbaArray('rgba(0,0,0,0.87)'),
+  secondary: colorToRgbaArray('rgba(0,0,0,0.54)'),
+  disabled: colorToRgbaArray('rgba(0,0,0,0.38)'),
+  hint: colorToRgbaArray('rgba(0,0,0,0.38)'),
+  icon: colorToRgbaArray('rgba(0,0,0,0.38)'),
+  divider: colorToRgbaArray('rgba(0,0,0,0.12)')
```

Comment:
Actually I think this just makes the theming more complicated as it needs to be.

We should just extend the foreground palettes, by creating those in contrast mode. 

We should be able to do this automatically (recommended) or manually.